### DISCUSSION TITLE
measuring waterpressure

### DISCUSSION THREAD

**Discussion by yann1420** [arendst/Tasmota#13407] [Q&A] [answered]
Hello all,
I live in a flat, 4th floor and sometimes the water pressure drops where taking a show is not really possible.
Therefore I would like to measure the current water pressure. That will help me to:
-not take a shower when there is not enough water
-make a fist against the water company  
My question to this forum if there is a known sensor that can do this and would tasmota support it?

Thank you

**Comment by sfromis** [comment, marked as answer]
On a general level, if you get a pressure transducer outputting an analog voltage value for pressure, it should be quite feasible to adapt the voltage, and calibrate analog values for pressure. This can be done without Tasmota having explicit support for a particular sensor. Can't recommend a particular transducer, but probably easiest to skip those which outputs millivolts.

**Comment by trlafleur** [comment]
see 
https://github.com/arendst/Tasmota/discussions/13310#discussioncomment-1452430

I use sensors like this to measure water pressure...

~~~
https://www.ebay.com/itm/373091627829?_trkparms=ispr%3D1&hash=item56ddfe7335:g:KacAAOSwykde4N04&amdata=enc%3AAQAGAAACoPYe5NmHp%252B2JMhMi7yxGiTJkPrKr5t53CooMSQt2orsSafTQYbq3L7RBVAMi0K9cwwt2v8GkjJTYurVS7lF1rzxXenhlOoGw%252F3JIWscEHBptGDZMnfxJVAyTWUkdPVqe4C0xHaLE0XHwbia8gh4zZTH7wQmCfoyswmVaTCAQFHmJOisf0w%252BDIu%252FDENpnr3XAJXFcWoT%252FTg3ZWe%252FuKbtyCIN1q0nlEsB3X5ogeOQfmmKYcbiCc%252B7J0p84OJK%252FMrExtyWHbCd8TFLhe4UzzGXS3PgZ3pNvpSlerkRld5BW44Lfv0cHnkDSLY916KjfXbnTqswcw3KiqHpGhEDTBDhhcDHTdDHQQ8m0TYJI71ynF611aqnybSoKhQlrkTAHyvqegd%252BeTbKtp9I9QCYxXBwbtzJmV5Mhh%252BRvhtqhFswZhWwg4Z9IlJRmR8F7IJsCwrgKN0YgUaoo%252BiW8r9ODKd8jmsuOyQs4jNJ2EHGwvzAJJzfz%252Fa5f7SmUAq5taLkoqjaeYYH1N2XdWa38bds%252FjKbB2WtTTTVJclew%252Bt6b3KUi%252Bxq3nAkvblAJ5nvzE5bPtWNXWLEf52du7Kuz7PwG9R38DivUp2uBpcEqC316dq6BCoVVi%252FEAgLEegtrfP8lZiSEquTcWcaOka%252BdAJlKu3BMFr31ym1%252FETUf%252BIwyc0%252FSC3w6YVbHQV8KXa3Qn5CBUMN3KUBsjpgvMSO1M5QD18a8%252BwfjmA4Z9DQPOc2tS1ez6Yu9u8RlcnrBv2GaDoypflyLr%252BuvdgHf%252FfO5HN%252BRFxU94Op8v0VVh%252FmRAw7Ltoxi6%252FERibYkZsh5Y%252FDf825Z%252BNxryV0FefBoZF4wTeMrG5fly7W%252BudfJ3eel0wIu0waOLAWk%252F06nb3e30wNghdKepYWGDGQ%253D%253D%7Cclp%3A2334524%7Ctkp%3ABFBMmJbHmZdf
~~~

**Reply by barbudor** [reply]
Voltage output : 
`Output:	 0.5V – 4.5V linear voltage output. 0 psi outputs 0`
Is you are using a NodeMCU or D1-Mini, the full range of the ADC input is 3.2V due to an existing on board voltage divider with a 220K and a 100K resitor. (ESP8266 native ADC full range is 1.0V)
You just need to insert a 150K resistor in serie with A0 input to get 4.7V full range
So you can use Tasmota ADC (GPIO17) in "ADC Range" mode
and use command `AdcParam 0, 979, 0, 100` to remap ADC range [0..979] to 0..100 PSI
979 = 1023 * 4.5/4.7

If using a bare ESP module, a voltage divider with 370K + 100K will give the same result

**Reply by mmiller1987** [reply]
I'm trying to do the same thing however I've got my pressure transducer connected to an ADS1115. The AdsParam commands seem to only work for the onboard A0 ADC pin. Any ideas on how to pass those commands to the specific channels of the i2c ADS1115?

**Reply by sfromis** [reply]
No way of doing that. You can have rules to get the data, and do your own calculations (like using the `Scale` command), which can be in addition to the standard analog values. Or you can do the processing backend.

**Reply by mmiller1987** [reply]
Thank you for clarifying. I will forward raw data to openhab or home assistant and transform there. As a follow up question: Can any of the tasmota commands interact with the data without rules?

**Reply by sfromis** [reply]
Not really but nevertheless..... While some basic commands like `Add`, `Sub` and `Scale` _can_ be used outside of a rule, you can only get the sensor data using rules. There also exists a scripting language as an alternative to rules, but that has more of a learning curve than rules. And with ESP32, you also have a "more normal" programming language in Berry.

**Reply by florinssk** [reply]
> Voltage output : `Output: 0.5V – 4.5V linear voltage output. 0 psi outputs 0` Is you are using a NodeMCU or D1-Mini, the full range of the ADC input is 3.2V due to an existing on board voltage divider with a 220K and a 100K resitor. (ESP8266 native ADC full range is 1.0V) You just need to insert a 150K resistor in serie with A0 input to get 4.7V full range So you can use Tasmota ADC (GPIO17) in "ADC Range" mode and use command `AdcParam 0, 979, 0, 100` to remap ADC range [0..979] to 0..100 PSI 979 = 1023 * 4.5/4.7
> 
> If using a bare ESP module, a voltage divider with 370K + 100K will give the same result


can you make a diagram, pls 
i'm not sure, that my english it's ok to understand. 
i want to use wemos d1.

**Comment by yann1420** [comment]
Thank you for the suggestions. I think I can bring this to a good end.
Again, thanks everyone

**Comment by javier6d2** [comment]
@barbudor, if output 0,5V is 0 PSI, would also necessary to remap ADC range low value with 979*0.5/4.5=108 ?

`AdcParam 6, 108, 979, 0, 100`

**Reply by sfromis** [reply]
Whatever would make it work. If you have two points, and know the pressure and analog value of each, AdcRange facilitates a linear fitting. Instead of starting from the voltage, using the analog reading allows more precise calibration.
Your `AdcParam 6, 108, 979, 0, 100` assumes that you get an analog value of 108 when the pressure is zero, and 979 when the pressure is 100.

**Comment by javier6d2** [comment]
Thank you for the clarification @sfromis. I have just tested and works very well, I won´t use any more because ADCrange is just an integer result and I would like to see any decimals, I work with BAR units instead PSI units, so I need. 
I will made the analog value conversion with a rule or scripting.

**Reply by sfromis** [reply]
In a rule, you have the Scale command, which is very similar to AdcRange, except for also allowing decimals. Of course, you could also stay with AdcParam, and set it up to calculate millibar (without a decimal point).
